### ISSUE TITLE
[Ascend] _topk_topp_kernel fails to compile — ConvertTritonIRToLinalgIR PassManager::run failed

### ISSUE THREAD

**Issue by Al2O3Al2O3** [flagos-ai/FlagTree#1220]
# [Ascend] `_topk_topp_kernel` fails to compile — `ConvertTritonIRToLinalgIR` PassManager::run failed

**Repo:** https://github.com/flagos-ai/FlagTree/

## Summary

vLLM's built-in Triton top-k/top-p sampling kernel (`_topk_topp_kernel` from
`vllm/v1/sample/ops/topk_topp_triton.py`) fails to compile on the Ascend
backend. The FlagTree ascend compiler aborts in the `ConvertTritonIRToLinalgIR`
stage with `PassManager::run failed`, which kills the vLLM worker → EngineCore →
the whole server.

This is the same class of failure MetaX already worked around: the FL MetaX
vendor patch (`vllm_fl/.../metax/patches/topk_topp_sampler.py`) explicitly forces
the PyTorch fallback with the comment *"vLLM's Triton topk_topp kernel fails to
compile on MetaX (PassManager::run failed in ttgir stage)"*. The Ascend backend
has no equivalent handling and hits the same wall.

## Environment

| Component | Version |
|---|---|
| flagtree | `0.7.0rc2+ascend3.5` |
| vllm | `0.20.2+flagos` |
| flag_gems | `5.4.0rc2.post3+gad89ec477` |
| torch / torch_npu | `2.10.0` |
| CANN | `9.0.0` |
| Target | `Ascend910_9382` (910C) |
| Image | `harbor.baai.ac.cn/flagos-app/vllm0.20.2-ascend-cann9.0.0-910c:2.1.2-0.2.2rc2.post1_gdb28502.d20260915` |

## Reproduction

1. Serve any model with the `fl` plugin on Ascend (TP=2):
   ```bash
   export VLLM_PLUGINS=fl VLLM_FL_PLATFORM=ascend
   vllm serve /models/Qwen3.6-35B-A3B --tensor-parallel-size 2 \
     --enforce-eager --max-model-len 4096 --gpu-memory-utilization 0.8
   ```
2. Send **≥ 8 concurrent** requests with sampling params (`top_k` and/or
   `top_p`, `temperature > 0`). vLLM routes to the Triton kernel only when
   `HAS_TRITON and logits.shape[0] >= 8` (see `apply_top_k_top_p` in
   `vllm/v1/sample/ops/topk_topp_sampler.py`), so the batch must reach 8.

## Observed error

```
File "/opt/flagtree/triton/spec/ascend/compiler/compiler.py", line 351, in compile
    raise MLIRCompilationError(stage_name, error_detail) from e
triton.compiler.errors.MLIRCompilationError:
///------------------[ERROR][Triton][BEG]------------------
[ConvertTritonIRToLinalgIR] encounters error:
PassManager::run failed
[INFO]: The compiled kernel cache is in /root/.triton/cache/F75RPTIC47EHHL4UAYSWJWU6UNDTUAO42AWIQTEM5CVI2TIP2BEA
///------------------[ERROR][Triton][END]------------------
```

Kernel source dumped by the compiler: `_topk_topp_kernel.source` / `.ttir`
(entry `tt.func public @_topk_topp_kernel`, target `Ascend910_9382`). The kernel
originates from `vllm/v1/sample/ops/topk_topp_triton.py:94`. Happy to attach the
full `.ttir` / `.source` on request.

## Impact

Any concurrent sampling workload (e.g. a benchmark at 8+ concurrency with
`top_k=20, top_p=0.95`) crashes the engine. Greedy decoding (`temperature=0`) is
unaffected because it never enters this kernel.

## Suggested fix

Fix the `ConvertTritonIRToLinalgIR` pass so this kernel lowers on Ascend. As an
interim mitigation the FL integration layer could add an Ascend vendor patch that
forces `apply_top_k_top_p` to the PyTorch path, mirroring the existing MetaX
patch — **but note** the PyTorch fallback on Ascend is itself broken (see the
companion issue on degenerate sampling output), so the mitigation alone is not
sufficient here.

**Comment by Al2O3Al2O3** [comment]
Cross-reference: the PyTorch fallback path (used at batch < 8, or if this Triton kernel is force-disabled) is itself broken on Ascend by a separate FlagGems bug — Tensor.exponential_() emits NaN for large tensors, see flagos-ai/FlagGems#6446. So mitigating this compile crash by forcing the PyTorch fallback is not sufficient on its own; both need fixing for random sampling to work on Ascend.
